launchlib/launcher.go

```diff
@@ -86,7 +86,33 @@ func compileCmdFromConfig(
 	var executable string
 	var executableErr error
 
-	if staticConfig.Type == "java" {
+	// Handle experimental nativeImage execution mode before standard processing
+	if customConfig.Experimental.ExecutionMode == ExecutionModeNative {
+		executable, executableErr = verifyPathIsSafeForExec(customConfig.Experimental.NativeImageExecutablePath)
+		if executableErr != nil {
+			return nil, executableErr
+		}
+		args = append(args, executable) // 0th argument is the command itself
+		// Add default system properties
+		args = append(args, "-Djava.io.tmpdir=var/data/tmp", "-Dsun.net.inetaddr.ttl=20")
+		for _, nativeArg := range customConfig.Experimental.NativeImageArguments {
+			args = append(args, nativeArg)
+		}
+		// If heapPercentage is specified, ensure MaxRAMPercentage flag is present
+		if customConfig.HeapPercentage != nil {
+			hasFlag := false
+			for _, arg := range customConfig.Experimental.NativeImageArguments {
+				if strings.HasPrefix(arg, "-XX:MaxRAMPercentage=") {
```

Comment:
The native flag is actually `-XX:MaximumHeapSizePercent` 
https://www.graalvm.org/latest/reference-manual/native-image/optimizations-and-performance/MemoryManagement/